[MJ]미션2: 체스판 생성

MJbae opened the pull request:
안녕하세요 MJ입니다.
미션 요구사항에 따라 구현해보았습니다.

Commit:
Fix: 요구사항에 따라 입력값 변경(이넘->문자열)

Commit:
Fix: 클래스 변수의 접근제어자 수정(private->private final)

Commit:
Refactor: 폰 클래스 기본생성자 적용

Commit:
Refactor: Color Enum 이동

Commit:
Test: BoardTest 객체 추가

Commit:
Add: 보드 객체 추가

Commit:
Chore: 폰, 폰테스트 클래스 패키지 이동

Commit:
Refactor: 테스트 코드 내 코드중복에 따른 메소드 분리

honux77 commented:
PR 본문에 했던 작업 내용을 요약하면 더 좋을 것 같아요.

honux77 commented on src/main/java/mj/chess/Pawn.java:
```diff
@@ -1,7 +1,7 @@
 
 public class Pawn {
-    private Color color;
+    private final Color color;
```
final도 나쁘지 않군요.

honux77 commented:
커밋 메시지를 가능하면 구체적으로 적어주세요. 제목만으로 무엇을 했는지 잘 알 수가 없네요.
그리고 커밋에는 Why?를 적어주시는 것이 좋습니다.

honux77 commented on src/main/java/mj/chess/pieces/Pawn.java:
```diff
@@ -0,0 +1,21 @@
+        this.color = Color.WHITE;
+    }
+
+    public Pawn(String color) {
```
enum을 사용했으면 생성자도 enum으로 하는 게 좋을 것 같아요.

honux77 commented on src/main/java/mj/chess/pieces/Pawn.java:
```diff
@@ -0,0 +1,21 @@
+        this.color = Color.valueOf(color.toUpperCase());
+    }
+
+    public String getColor() {
```
여기도 enum으로 리턴

MJbae replied:
enum을 어설프게 공부한거 같아요. 제대로 공부해서 수정해보겠습니다! 리뷰 감사합니다.

honux77 commented on src/test/java/mj/chess/BoardTest.java:
```diff
@@ -0,0 +1,30 @@
+    @Test
+    @DisplayName("보드 객체 내 흰색 폰 1개, 검은색 폰 1개 생성")
+    public void create_board() {
+        check_validity_of_pawn(new Pawn(WHITE.name()), 1, 0);
```
가능하면 스네이크 케이스보다는 카멜 케이스가 자바에 잘 어울립니다.

honux77 commented on src/test/java/mj/chess/BoardTest.java:
```diff
@@ -0,0 +1,30 @@
+import static mj.chess.pieces.Pawn.Color.*;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public class BoardTest {
```
junit5에서 테스트 메소드는 접근제어자 없이 사용하는 게 일반적입니다.

MJbae replied:
그렇군요! 적용하겠습니다. 리뷰 감사합니다.

honux77 reviewed (changes requested):
enum을 사용하면 type safe가 생기는데요. 해당 부분 공부하고  String으로 처리한 부분을 enum으로 바꾸면 좋겠네요.

Commit:
Refactor: Pawn class의 생성자 매개변수로 Enum 값 전달

기존에는 Pawn class의 생성자 매개변수로 String을 전달하고 생성자 내부에서 매개변수로 받은 값을 제한했음
하지만 코드의 간결성을 고려하여 매개변수로 Enum값을 전달하여 매개변수의 타입을 제한함

Commit:
Refactor: 컨벤션에 따라 메소드명을 카멜케이스로 수정

Commit:
Refactor: 테스트 클래스 및 메소드의 접근제어자 생략(public -> default)

JUnit5에서부터 기본 접근제어자를 default로 사용할 것을 권장, 가독성을 향상할 수 있음

Commit:
Refactor: createBoard 테스트 메소드 내 내부 메소드를 두 개로 분할

메소드 하나에 두 개의 기능이 포함되었기 때문에 각각의 기능에 대해 메소드로 재정의

MJbae commented:
@honux77 코멘트 남겨주신 부분을 반영해서 리팩토링했습니다~

honux77 commented on build.gradle:
```diff
@@ -9,6 +9,7 @@ repositories {
 }
 
 dependencies {
+    implementation 'junit:junit:4.12'
```
junit4 의존성이 들어갔는데 제거해 주세요.

honux77 replied:
커밋하기 전에 변경사항을 한 번 더 확인해 주는 것도 잊지 마세요.

honux77 commented on src/main/java/mj/chess/Board.java:
```diff
@@ -7,8 +7,7 @@
 public class Board {
     private ArrayList<Pawn> pawns = new ArrayList<>();
```
앗 이것도 놓쳤는데 `List<Pawn> pawns` 로 선언하는게 일반적입니다.

honux77 commented on src/main/java/mj/chess/pieces/Pawn.java:
```diff
@@ -1,21 +1,18 @@
 
 public class Pawn {
-    private final Color color;
+    private final ColorOfPiece color;
```
Color를 ColorOfPiece로 바꿨는데 왜 그랬는지는 잘 모르겠네요.

honux77 commented on src/test/java/mj/chess/BoardTest.java:
```diff
@@ -1,30 +1,37 @@
     }
 
-    public void check_validity_of_pawn(Pawn pawn, int size, int index){
+    void verifyAddition(Pawn pawn, int size){
```
테스트 내부에서 사용하는 메소드는 private이 좋을 듯

MJbae replied:
다른 그룹원 리뷰에서 봤던 내용인데도 까먹고 적용을 못했네요... 코멘트 감사합니다.

honux77 reviewed (approved):
enum을 그냥 기존처럼 Color로 선언하는게 더 자연스러워 보입니다.
몇가지 코멘트는 다음 PR에 함께 반영해 주세요.
